src/Uno.UI/UI/Xaml/ResourceResolver.cs

```diff
@@ -405,14 +405,17 @@ private static bool TryAssemblyResourceRetrieval(in SpecializedResourceDictionar
 				return false;
 			}
 
-			if (_registeredDictionariesByAssembly.TryGetValue(parseContext.AssemblyName, out var assemblyDict))
+			if (parseContext.AssemblyName is not null)
 			{
-				foreach (var kvp in assemblyDict)
+				if (_registeredDictionariesByAssembly.TryGetValue(parseContext.AssemblyName, out var assemblyDict))
 				{
-					var rd = kvp.Value as ResourceDictionary;
-					if (rd.TryGetValue(resourceKey, out value, shouldCheckSystem: false))
+					foreach (var kvp in assemblyDict)
 					{
-						return true;
+						var rd = kvp.Value as ResourceDictionary;
+						if (rd.TryGetValue(resourceKey, out value, shouldCheckSystem: false))
```

Comment:
In this scenario where we don't care about the `out value` isn't more convenient to use `ContainsKey` method?

Reply:
I think we care about the out value. That is reflected in the out parameter of this method

Reply:
Indeed, the out value is used as the out value of the containing method.

Reply:
I see it, thanks, I missed it the first time. :)